katdal/datasources.py

```diff
@@ -131,18 +135,50 @@ def name(self):
         return name
 
 
+def view_capture_stream(telstate, capture_block_id=None, stream_name=None):
+    """Create telstate view based on detected capture block ID and stream name."""
+    # Detect the capture block
+    if not capture_block_id:
+        capture_blocks = []
+        if 'sdp_capture_block_id' in telstate:
+            for value_time in telstate.get_range('sdp_capture_block_id', st=0):
+                cbid = value_time[0]
+                if 'obs_params' in telstate.view(cbid, exclusive=True):
+                    capture_blocks.append(cbid)
+        if not capture_blocks:
+            raise ValueError('No capture block IDs found in telstate - '
+                             'please specify it manually')
+        # Pick the latest capture block
+        capture_block_id = capture_blocks[-1]
+        if len(capture_blocks) > 1:
+            logger.warn('Telstate has more than one capture block - {} - '
+                        'picking the latest'.format(capture_blocks))
```

Comment:
`warn` is [deprecated](https://docs.python.org/3/library/logging.html?highlight=warning#logging.warning) in favour of `warning`. Also do logger formatting with %d and *args.

Reply:
Dang, I always used that because it is shorter (gotta cram those log messages into 79 lines!).

Reply:
Done.